docs/combinatorics.rst

```diff
@@ -1,6 +1,202 @@
 .. _sec_combinatorics:
 
-=====================================
-Ranking and Unranking Tree Topologies
-=====================================
-TODO
+=============
+Combinatorics
+=============
+This module implements a combinatorial approach to counting topologies
```

Comment:
We're not implementing it as an externally visible module - can you change the intro to talking about tskit in general rather than the combinatorics module?

Reply:
Do you want to keep the generating functions in the combinatorics module, or put them under a different namespace? That would affect whether the module needs to be visible.

Reply:
We can make them visible externally by exporting from ``__init__.py``, like 
```
from . combinatorics import all_tree_topologies 
```
 etc. (Maybe this is a better name than ``all_unlabelled_trees``?)

Reply:
> We can make them visible externally by exporting from `__init__.py`, like

Sounds good (I'm flattered my functions get top-level)

> etc. (Maybe this is a better name than all_unlabelled_trees?)

I think we've been referring to topologies without further specification as the leaf-labelled trees. Based on what I have now, we could do:

```
all_tree_topologies(n) (right now this is all_trees)
all_tree_shapes(n)
all_tree_labellings(t)
```

or could switch it to something like:

```
all_labelled_tree_topologies(n)
all_tree_topologies(n)
all_tree_labellings(t)?
```